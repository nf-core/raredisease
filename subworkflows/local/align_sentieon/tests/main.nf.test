nextflow_workflow {

    name "Test Workflow ALIGN_SENTIEON"
    script "subworkflows/local/align_sentieon/main.nf"
    workflow "ALIGN_SENTIEON"
    tag "subworkflows"
    tag "align_sentieon"
    config "./nextflow.config"

    test("align sentieon") {

        setup {
            run("SENTIEON_BWAINDEX") {
                script "modules/nf-core/sentieon/bwaindex/main.nf"
                process {
                    """
                    input[0] = Channel.of([
                        [id:'sarscov2'],
                        file('https://raw.githubusercontent.com/nf-core/test-datasets/modules/data/genomics/sarscov2/genome/genome.fasta', checkIfExists: true)
                    ])
                    """
                }
            }
        }

        when {
            params {
                aligner = "sentieon"
                sarscov_testdata_base_path= 'https://raw.githubusercontent.com/nf-core/test-datasets/modules/data/'
                save_mapped_as_cram = true
            }
            workflow {
                """
                input[0] = Channel.of([
                    [ id:'test', sample:'test', single_end:false, num_lanes:1, read_group: "'@RG\\\\tID:test\\\\tPL:illumina\\\\tSM:test'"  ], // meta map
                    [
                        file(params.sarscov_testdata_base_path+ 'genomics/sarscov2/illumina/fastq/test_1.fastq.gz', checkIfExists: true),
                        file(params.sarscov_testdata_base_path+ 'genomics/sarscov2/illumina/fastq/test_2.fastq.gz', checkIfExists: true)
                    ]
                ])
                input[1] = Channel.of([[id:'sarscov2'], [file(params.sarscov_testdata_base_path+ 'genomics/sarscov2/genome/genome.fasta', checkIfExists: true)]])
                input[2] = Channel.of([[id:'sarscov2'], [file(params.sarscov_testdata_base_path+ 'genomics/sarscov2/genome/genome.fasta.fai', checkIfExists: true)]])
                input[3] = SENTIEON_BWAINDEX.out.index
                input[4] = "illumina"
                """
            }
        }

        then {
            assert workflow.success
            assert snapshot(workflow.out).match()
        }

    }

}
