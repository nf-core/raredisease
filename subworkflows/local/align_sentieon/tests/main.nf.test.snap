{
    "align sentieon": {
        "content": [
            {
                "0": [
                    [
                        {
                            "groupSize": 1,
                            "groupTarget": {
                                "id": "test",
                                "sample": "test",
                                "single_end": false,
                                "num_lanes": 1,
                                "read_group": "'@RG\\tID:test\\tPL:illumina\\tSM:test'"
                            }
                        },
                        "test_dedup.bam:md5,0dda7e8e144c07591f06330b07b36071"
                    ]
                ],
                "1": [
                    [
                        {
                            "groupSize": 1,
                            "groupTarget": {
                                "id": "test",
                                "sample": "test",
                                "single_end": false,
                                "num_lanes": 1,
                                "read_group": "'@RG\\tID:test\\tPL:illumina\\tSM:test'"
                            }
                        },
                        "test_dedup.bam.bai:md5,0458a0ef98b3a86ba28a2bcac628f7d4"
                    ]
                ],
                "2": [
                    [
                        {
                            "groupSize": 1,
                            "groupTarget": {
                                "id": "test",
                                "sample": "test",
                                "single_end": false,
                                "num_lanes": 1,
                                "read_group": "'@RG\\tID:test\\tPL:illumina\\tSM:test'"
                            }
                        },
                        "test_datametrics_mq_metrics.txt:md5,1ba9b952749ad4e5310598fe23c02085"
                    ]
                ],
                "3": [
                    [
                        {
                            "groupSize": 1,
                            "groupTarget": {
                                "id": "test",
                                "sample": "test",
                                "single_end": false,
                                "num_lanes": 1,
                                "read_group": "'@RG\\tID:test\\tPL:illumina\\tSM:test'"
                            }
                        },
                        "test_datametrics_qd_metrics.txt:md5,5294d174c6ffaaf47ceb02befbbffc40"
                    ]
                ],
                "4": [
                    [
                        {
                            "groupSize": 1,
                            "groupTarget": {
                                "id": "test",
                                "sample": "test",
                                "single_end": false,
                                "num_lanes": 1,
                                "read_group": "'@RG\\tID:test\\tPL:illumina\\tSM:test'"
                            }
                        },
                        "test_datametrics_gc_metrics.txt:md5,96c951444c7f5af19c60d5262e74fb1c"
                    ]
                ],
                "5": [
                    [
                        {
                            "groupSize": 1,
                            "groupTarget": {
                                "id": "test",
                                "sample": "test",
                                "single_end": false,
                                "num_lanes": 1,
                                "read_group": "'@RG\\tID:test\\tPL:illumina\\tSM:test'"
                            }
                        },
                        "test_datametrics_gc_summary.txt:md5,77045bf67dbc251523af0d5bf9b57dbf"
                    ]
                ],
                "6": [
                    [
                        {
                            "groupSize": 1,
                            "groupTarget": {
                                "id": "test",
                                "sample": "test",
                                "single_end": false,
                                "num_lanes": 1,
                                "read_group": "'@RG\\tID:test\\tPL:illumina\\tSM:test'"
                            }
                        },
                        "test_datametrics_aln_metrics.txt:md5,f4681d1f08b687f424af70772cfb597a"
                    ]
                ],
                "7": [
                    [
                        {
                            "groupSize": 1,
                            "groupTarget": {
                                "id": "test",
                                "sample": "test",
                                "single_end": false,
                                "num_lanes": 1,
                                "read_group": "'@RG\\tID:test\\tPL:illumina\\tSM:test'"
                            }
                        },
                        "test_datametrics_is_metrics.txt:md5,90bfb2b94b01074d93e6da1b69c4e494"
                    ]
                ],
                "8": [
                    "versions.yml:md5,01896e6794bd10730ddfe6872631bdde",
                    "versions.yml:md5,2b8ad96b3aa67e6563e8a05744db6174",
                    "versions.yml:md5,c0917d5569d12b01718342f7aac5cbbd"
                ],
                "aln_metrics": [
                    [
                        {
                            "groupSize": 1,
                            "groupTarget": {
                                "id": "test",
                                "sample": "test",
                                "single_end": false,
                                "num_lanes": 1,
                                "read_group": "'@RG\\tID:test\\tPL:illumina\\tSM:test'"
                            }
                        },
                        "test_datametrics_aln_metrics.txt:md5,f4681d1f08b687f424af70772cfb597a"
                    ]
                ],
                "gc_metrics": [
                    [
                        {
                            "groupSize": 1,
                            "groupTarget": {
                                "id": "test",
                                "sample": "test",
                                "single_end": false,
                                "num_lanes": 1,
                                "read_group": "'@RG\\tID:test\\tPL:illumina\\tSM:test'"
                            }
                        },
                        "test_datametrics_gc_metrics.txt:md5,96c951444c7f5af19c60d5262e74fb1c"
                    ]
                ],
                "gc_summary": [
                    [
                        {
                            "groupSize": 1,
                            "groupTarget": {
                                "id": "test",
                                "sample": "test",
                                "single_end": false,
                                "num_lanes": 1,
                                "read_group": "'@RG\\tID:test\\tPL:illumina\\tSM:test'"
                            }
                        },
                        "test_datametrics_gc_summary.txt:md5,77045bf67dbc251523af0d5bf9b57dbf"
                    ]
                ],
                "is_metrics": [
                    [
                        {
                            "groupSize": 1,
                            "groupTarget": {
                                "id": "test",
                                "sample": "test",
                                "single_end": false,
                                "num_lanes": 1,
                                "read_group": "'@RG\\tID:test\\tPL:illumina\\tSM:test'"
                            }
                        },
                        "test_datametrics_is_metrics.txt:md5,90bfb2b94b01074d93e6da1b69c4e494"
                    ]
                ],
                "marked_bai": [
                    [
                        {
                            "groupSize": 1,
                            "groupTarget": {
                                "id": "test",
                                "sample": "test",
                                "single_end": false,
                                "num_lanes": 1,
                                "read_group": "'@RG\\tID:test\\tPL:illumina\\tSM:test'"
                            }
                        },
                        "test_dedup.bam.bai:md5,0458a0ef98b3a86ba28a2bcac628f7d4"
                    ]
                ],
                "marked_bam": [
                    [
                        {
                            "groupSize": 1,
                            "groupTarget": {
                                "id": "test",
                                "sample": "test",
                                "single_end": false,
                                "num_lanes": 1,
                                "read_group": "'@RG\\tID:test\\tPL:illumina\\tSM:test'"
                            }
                        },
                        "test_dedup.bam:md5,0dda7e8e144c07591f06330b07b36071"
                    ]
                ],
                "mq_metrics": [
                    [
                        {
                            "groupSize": 1,
                            "groupTarget": {
                                "id": "test",
                                "sample": "test",
                                "single_end": false,
                                "num_lanes": 1,
                                "read_group": "'@RG\\tID:test\\tPL:illumina\\tSM:test'"
                            }
                        },
                        "test_datametrics_mq_metrics.txt:md5,1ba9b952749ad4e5310598fe23c02085"
                    ]
                ],
                "qd_metrics": [
                    [
                        {
                            "groupSize": 1,
                            "groupTarget": {
                                "id": "test",
                                "sample": "test",
                                "single_end": false,
                                "num_lanes": 1,
                                "read_group": "'@RG\\tID:test\\tPL:illumina\\tSM:test'"
                            }
                        },
                        "test_datametrics_qd_metrics.txt:md5,5294d174c6ffaaf47ceb02befbbffc40"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,01896e6794bd10730ddfe6872631bdde",
                    "versions.yml:md5,2b8ad96b3aa67e6563e8a05744db6174",
                    "versions.yml:md5,c0917d5569d12b01718342f7aac5cbbd"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-04-09T14:36:45.125659653"
    }
}